packages/squiggle-lang/src/public/SqValue/SqLambda.ts

```diff
@@ -21,26 +17,27 @@ export type SqLambdaParameter = {
   typeName?: string;
 };
 
-function lambdaToSqLambdaParameters(lambda: Lambda): SqLambdaParameter[][] {
-  if (lambda instanceof UserDefinedLambda) {
-    return [
-      lambda.parameters.map((param) => {
-        return {
-          name: param.name,
-          domain: param.domain ? wrapDomain(param.domain.value) : undefined,
-        };
-      }),
-    ];
-  } else if (lambda instanceof BuiltinLambda) {
-    return lambda.definitions().map((def) =>
-      def.map((p, index) => ({
-        name: index.toString(),
-        domain: undefined,
-        typeName: p.getName(),
-      }))
-    );
-  } else {
-    throw new Error("Unknown lambda type");
+type SqLambdaSignature = SqLambdaParameter[];
+
+function lambdaToSqLambdaSignatures(lambda: Lambda): SqLambdaSignature[] {
+  switch (lambda.type) {
+    case "UserDefinedLambda":
+      return [
+        lambda.parameters.map((param) => {
+          return {
+            name: param.name,
+            domain: param.domain ? wrapDomain(param.domain.value) : undefined,
+          };
+        }),
+      ];
+    case "BuiltinLambda":
+      return lambda.signatures().map((def) =>
+        def.map((p, index) => ({
+          name: index.toString(),
+          domain: undefined,
+          typeName: p.getName(),
+        }))
+      );
```

Comment:
Ah, right, if your function has an explicit return type then you don't even need `satisfies never` check.